integration/securitypolicy_test.go

```diff
@@ -0,0 +1,400 @@
+		Rate:   34,
+		Active: true,
+		Tags:   []string{"testing"},
+	}
+
+	hasSameValues := func(k8sSpec, tykSpec *v1alpha1.SecurityPolicySpec, k8sStatusID string) bool {
+		return k8sSpec.MID == tykSpec.MID &&
+			len(k8sSpec.Tags) == len(tykSpec.Tags) &&
+			len(tykSpec.Tags) == 1 &&
+			tykSpec.Tags[0] == initialK8sPolicyTag &&
+			tykSpec.Rate == initialK8sPolicyRate &&
+			tykSpec.State == initialK8sPolicyState
+	}
+
+	var (
+		polRec   controllers.SecurityPolicyReconciler
+		policyCR v1alpha1.SecurityPolicy
+		reqCtx   context.Context
+	)
+
+	securityPolicyMigrationFeatures := features.New("Existing Security Policy Migration to K8s").
+		Setup(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			eval := is.New(t)
+
+			opConfSecret := v1.Secret{}
+			err := c.Client().Resources(opNs).Get(ctx, "tyk-operator-conf", opNs, &opConfSecret)
+			eval.NoErr(err)
+
+			// Obtain Environment configuration to be able to connect Tyk.
+			tykEnv, err := generateEnvConfig(&opConfSecret)
+			eval.NoErr(err)
+
+			if tykEnv.Mode == "ce" {
+				t.Skip("SecurityPolicy API is not implemented in CE yet")
+			}
+
+			testCl, err := createTestClient(c.Client())
+			eval.NoErr(err)
+
+			polRec = controllers.SecurityPolicyReconciler{
+				Client: testCl,
+				Log:    log.NullLogger{},
+				Scheme: testCl.Scheme(),
+				Env:    tykEnv,
+			}
+
+			reqCtx = tykClient.SetContext(context.Background(), tykClient.Context{
+				Env: polRec.Env,
+				Log: log.NullLogger{},
+			})
+
+			return ctx
+		}).
+		Assess("Migrate a simple Security Policy from Dashboard to K8s",
+			func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+				eval := is.New(t)
+
+				testNs, ok := ctx.Value(ctxNSKey).(string)
+				eval.True(ok)
+
+				err := klient.Universal.Portal().Policy().Create(reqCtx, &spec)
+				eval.NoErr(err)
+
+				policyCR = v1alpha1.SecurityPolicy{
+					ObjectMeta: metav1.ObjectMeta{Name: "sample-policy", Namespace: testNs},
+					Spec: v1alpha1.SecurityPolicySpec{
+						ID:     spec.MID,
+						Active: true,
+						State:  initialK8sPolicyState,
+						Rate:   initialK8sPolicyRate,
+						Tags:   []string{initialK8sPolicyTag},
+					},
+				}
+
+				_, err = util.CreateOrUpdate(ctx, polRec.Client, &policyCR, func() error {
+					return nil
+				})
+				eval.NoErr(err)
+
+				err = wait.For(func() (done bool, err error) {
+					_, err = polRec.Reconcile(ctx, ctrl.Request{NamespacedName: cr.ObjectKeyFromObject(&policyCR)})
+					return err == nil, err
+				})
+				eval.NoErr(err)
+
+				err = wait.For(
+					conditions.New(c.Client().Resources()).ResourceMatch(&policyCR, func(object k8s.Object) bool {
+						policyOnK8s, ok := object.(*v1alpha1.SecurityPolicy)
+						eval.True(ok)
+						eval.True(len(policyOnK8s.Status.PolID) > 0)
+
+						policyOnTyk, err := klient.Universal.Portal().Policy().Get(reqCtx, policyOnK8s.Status.PolID)
+						eval.NoErr(err)
+
+						eval.True(policyOnK8s.Spec.MID == policyOnTyk.MID &&
```

Comment:
Can't we use `hasSameValues()` here too?

Reply:
yes, we can. thanks for the suggestion. updated as you suggested in d17880a1cd921024b5b82813d7c62c2021ebb2cd